Purchases/Public/RCPurchases.m

```diff
@@ -219,32 +209,53 @@ - (void)cachePurchaserInfo:(RCPurchaserInfo *)info {
     }
 }
 
-- (void)updateCaches
+- (void)updateCaches {
+    [self updateCachesWithCompletionBlock:nil];
+}
+
+- (void)updateCachesWithCompletionBlock:(RCReceivePurchaserInfoBlock _Nullable)completion
 {
     NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
     if (self.cachesLastUpdated != nil && timeSinceLastCheck < 60.) {
-        [self readPurchaserInfoFromCache];
         return;
     }
 
     self.cachesLastUpdated = [NSDate date];
 
     [self.backend getSubscriberDataWithAppUserID:self.appUserID
-                                      completion:^(RCPurchaserInfo * _Nullable info,
-                                                   NSError * _Nullable error) {
-        if (error == nil) {
-            [self handleUpdatedPurchaserInfo:info error:nil];
-        } else {
-            self.cachesLastUpdated = nil;
-        }
-    }];
-
-    [self getEntitlements:^(NSDictionary<NSString *,RCEntitlement *> *entitlements) {}];
-}
-
--(NSDictionary<NSString *, RCEntitlement *> * _Nullable)entitlements
-{
-    return self.cachedEntitlements;
+                                 completion:^(RCPurchaserInfo * _Nullable info,
+                                              NSError * _Nullable error) {
+                                     if (error == nil) {
+                                         [self handleUpdatedPurchaserInfo:info error:nil];
+                                     } else {
+                                         self.cachesLastUpdated = nil;
+                                     }
+                                     
+                                     if (completion) {
+                                         [self dispatch:^{
+                                             completion(info, error);
+                                         }];
+                                     }
+                                 }];
+
+    [self getEntitlements:nil];
+}
+
+- (void)purchaserInfoWithCompletionBlock:(RCReceivePurchaserInfoBlock)completion
+{
+    RCPurchaserInfo *info = [self readPurchaserInfoFromCache];
```

Comment:
I think there should be a check to verify the caches are valid. The check can be done inside `readPurchaserInfoFromCache` and return `nil` if the cached version is too old.